lib/client/src/client.rs

```diff
@@ -779,65 +782,10 @@ impl MangoClient {
             Serum3Side::Ask => (&base, &quote),
         };
 
-        let mut open_orders_opt = account
+        let open_orders = account
             .serum3_orders(market_index)
             .map(|x| x.open_orders)
-            .ok();
-
-        let mut missing_tokens = false;
-
-        let token_replace_ixs = self
-            .find_existing_or_try_to_replace_token_positions(
-                &mut account,
-                Vec::from([payer_token.token_index, receiver_token.token_index]),
-            )
-            .await;
-        match token_replace_ixs {
-            Ok(res) => {
-                if let Some(ix) = res {
-                    ixs.append(ix);
-                }
-            }
-            Err(_) => missing_tokens = true,
-        }
-
-        if open_orders_opt.is_none() {
-            let has_available_slot = account.all_serum3_orders().any(|p| !p.is_active());
-            let should_close_one_open_orders_account = !has_available_slot || missing_tokens;
-
-            if should_close_one_open_orders_account {
-                ixs.append(
-                    self.deactivate_first_active_unused_serum3_orders(&mut account)
-                        .await?,
-                );
-            }
-
-            // in case of missing token slots
-            // try again to create, as maybe deactivating the market slot resulted in some token being now unused
-            // but this time, in case of error, propagate to caller
-            if missing_tokens {
-                if let Some(ix) = self
-                    .find_existing_or_try_to_replace_token_positions(
-                        &mut account,
-                        Vec::from([payer_token.token_index, receiver_token.token_index]),
-                    )
-                    .await?
-                {
-                    ixs.append(ix);
-                }
-            }
-
-            ixs.push(
-                self.serum3_create_open_orders_instruction(market_index),
-                self.context.compute_estimates.health_cu_per_serum,
-            );
-
-            let created_open_orders = self.serum3_create_open_orders_address(market_index);
-            open_orders_opt = Some(created_open_orders);
-            account.create_serum3_orders(market_index)?.open_orders = created_open_orders;
-        }
-
-        let open_orders = open_orders_opt.expect("oo is created");
+            .expect("oo is created");
```

Comment:
don't use `expect()` unless you want to crash on error. This would be better with `?` I think

Reply:
that's how it was before, but I can change it if you think it's better

Reply:
would prefer if you did, but then it's optional